jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -0,0 +1,374 @@
+            self.ext_trace_pixel = half_pixel
+        if self.ext_width == 0:
+            self.ext_width = int(np.ceil(width / 10))
+
+    @observe('plugin_opened')
+    def _update_plugin_marks(self, *args):
+        if self.active_step == '':
+            # on load, default to 'trace' (this will then trigger the observe to update the marks)
+            self.active_step = 'trace'
+            self._interaction_in_trace_step()
+            return
+
+        display_marks = {'trace': ['trace'],
+                         'bg': ['trace',
+                                'bg1_center', 'bg1_lower', 'bg1_upper',
+                                'bg2_center', 'bg2_lower', 'bg2_upper'],
+                         'ext': ['trace', 'ext_upper', 'ext_lower']}
+        for step, mark in self.marks.items():
+            mark.visible = self.plugin_opened and step in display_marks.get(self.active_step, [])
+
+    @property
+    def marks(self):
+        if self._marks:
+            # TODO: replace with cache property?
+            return self._marks
+
+        viewer = self.app.get_viewer('spectrum-2d-viewer')
+        if not viewer.state.reference_data:
+            # we don't have data yet for scales, defer initializing
+            return {}
+
+        # then haven't been initialized yet, so initialize with empty
+        # marks that will be populated once the first analysis is done.
+        self._marks = {'trace': TraceLine(viewer, visible=self.plugin_opened),
+                       'ext_lower': TraceLine(viewer, visible=self.plugin_opened),
+                       'ext_upper': TraceLine(viewer, visible=self.plugin_opened),
+                       'bg1_center': TraceLine(viewer, visible=self.plugin_opened, line_style='dotted'),
+                       'bg1_lower': TraceLine(viewer, visible=self.plugin_opened),
+                       'bg1_upper': TraceLine(viewer, visible=self.plugin_opened),
+                       'bg2_center': TraceLine(viewer, visible=self.plugin_opened, line_style='dotted'),
+                       'bg2_lower': TraceLine(viewer, visible=self.plugin_opened),
+                       'bg2_upper': TraceLine(viewer, visible=self.plugin_opened)}
+        shadows = [ShadowLine(mark, shadow_width=2) for mark in self._marks.values()]
+        # NOTE: += won't trigger the figure to notice new marks
+        viewer.figure.marks = viewer.figure.marks + shadows + list(self._marks.values())
+
+        return self._marks
+
+    @observe('trace_trace_selected', 'trace_offset', 'trace_dataset_selected', 'trace_type_selected',
+             'trace_pixel', 'trace_peak_method_selected', 'trace_bins', 'trace_window')
+    def _interaction_in_trace_step(self, *args):
+        if not self.plugin_opened:
+            return
+        trace = self.create_trace(add_data=False)
+        # TODO: range(len(...)) is a temporary hack for this data displaying in meters instead of pixels
+        self.marks['trace'].update_xy(range(len(trace.trace)), trace.trace)
```

Comment:
will definitely want to fix this or find a workaround so that the underlying bug doesn't become a blocker to this PR [:cat:](https://jira.stsci.edu/browse/JDAT-2581)